### PR TITLE
Track broken images

### PR DESCRIPTION
**Pull request by callumacrae**
https://trello.com/c/JN9KbZ15/754-error-tracking-from-monkey-for-cp

### CODE REVIEW

**Comment by samjbmason** [comment]
lgtm 👍

**Comment by djgrant** [comment]
👍  grouping and spacing is a bit inconsistent if you want me to be _really_ picky

**Comment by callumacrae** [comment]
I don't want you to be really picky

File: src/js/steps/initMonkey.js [hunk 1]
```diff
@@ -21,6 +23,26 @@ module.exports = function ($events, options) {
 
     data.monkeyContainer.addClass('ready');
 
+    // Track errors on image load
+    var $images = data.monkeyContainer.find('[class*="page"] img');
+    $images.on('error', function () {
+      if (window.analytics) {
+        analytics.track('Broken image in monkey', { src: this.src });
```
Comment (jelmersnoeck) [inline src/js/steps/initMonkey.js]: Would it be possible to add `pageID` to this as well (like `src`)? Would make it easier to filter.

File: src/js/steps/initMonkey.js [hunk 2]
```diff
@@ -21,6 +23,26 @@ module.exports = function ($events, options) {
...
+
+    var remaining = $images.filter(function () {
+      return !this.complete;
+    }).length;
+
+    $images.on('load', function () {
+      remaining--;
+
+      if (!remaining && window.analytics && options.platformAPI) {
+        analytics.track('Monkey images all loaded');
```
Comment (jelmersnoeck) [inline src/js/steps/initMonkey.js]: Not sure how flexible this system is, but would it be possible to time how long it took to load them all? And maybe individual pages?

Comment (callumacrae) [inline src/js/steps/initMonkey.js]: It's possible, but I don't think it'll be useful data. The slowest cases will be users on bad connections, not the creative platform being slow. Also, we intentionally delay the loading of some images.

File: src/js/steps/initMonkey.js (hunk 1 above)
Comment (callumacrae) [inline src/js/steps/initMonkey.js]: What would `pageID` be? The page number?

Comment (tonydaly) [inline src/js/steps/initMonkey.js]: Yeah `pageID` would be super useful.

**Comment by tonydaly** [comment]
@callumacrae looks good!

There's another step afterwards where we'll need to get the data out of Segment and into our own graphing, let's have a chat about it later.

**Comment by callumacrae** [comment]
@jelmersnoeck @tonydaly Added pageID to analytics call :)

**Pull request merged**